setup.py

```diff
@@ -974,8 +974,53 @@ def print_box(msg):
         print('|{}{}|'.format(l, ' ' * (size - len(l))))
     print('-' * (size + 2))
 
+def pre_build_checks():
+    if sys.version_info >= (3, 11):
```

Comment:
Also, as dynamo right now is not supported on Windows, shouldn't we return here if platform.name() is `win32`?

Reply:
Ho it is not? It is definitely getting compiled on windows (it have seen in fail in CI on earlier version of this PR). So I think we want to keep that check.